iptables/iptables.go

```diff
@@ -149,15 +151,51 @@ func addRulesForInboundPortRedirect(firewallConfiguration FirewallConfiguration,
 	return commands
 }
 
-func addRulesForIgnoredPorts(portsToIgnore []int, chainName string, commands []*exec.Cmd) []*exec.Cmd {
-	for _, ignoredPort := range portsToIgnore {
-		log.Printf("Will ignore port %d on chain %s", ignoredPort, chainName)
-
-		commands = append(commands, makeIgnorePort(chainName, ignoredPort, fmt.Sprintf("ignore-port-%d", ignoredPort)))
+func addRulesForIgnoredPorts(portsToIgnore []string, chainName string, commands []*exec.Cmd) []*exec.Cmd {
+	for _, destinations := range makeMultiportDestinations(portsToIgnore) {
+		log.Printf("Will ignore port(s) %s on chain %s", destinations, chainName)
+		commands = append(commands, makeIgnorePorts(chainName, destinations, fmt.Sprintf("ignore-port-%s", destinations)))
 	}
 	return commands
 }
 
+func makeMultiportDestinations(portsToIgnore []string) [][]string {
+	destinationSlices := make([][]string, 0)
+	destinationPortCount := 0
+	if portsToIgnore == nil || len(portsToIgnore) < 1 {
+		return destinationSlices
+	}
+	destinations := make([]string, 0)
+	for _, portOrRange := range portsToIgnore {
+		if strings.Contains(portOrRange, "-") {
+			if portRange, err := ports.ParsePortRange(portOrRange); err == nil {
+				if destinationPortCount+2 > 15 {
```

Comment:
Maybe its worth extracing and naming this threshold constant for better clarity (15) and also adding a comment about the rationale behind it.

Reply:
Extracted "15" as a constant and provided some comments around the logic.  Cleaned up the logic a bit to consolidate the handling of port ranges and single ports given other requested changes.